File: packages/transform/src/features/json/JsonCreateValidateParseTransformer.ts
import { JsonValidateParseProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace JsonCreateValidateParseTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "json.createValidateParse",
      write: JsonValidateParseProgrammer.write,
    });
}

File: packages/transform/src/features/json/JsonCreateValidateParseTransformer.go
package json

import internal "github.com/samchon/typia/packages/transform/src/internal"

var JsonCreateValidateParseTransformer = jsonCreateValidateParseTransformerNamespace{}

type jsonCreateValidateParseTransformerNamespace struct{}

func (jsonCreateValidateParseTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return `((input) => { try { return { success: true, data: JSON.parse(input) }; } catch (error) { return { success: false, errors: [{ path: "$input", expected: "json", value: input }] }; } })`
}
